File: programowanie obiektowe/2025.03.28/c31.py
def something():
    ff = Foo("Krzysio")

class Foo:
    def __init__(self, name):
        self.name = name
        print(f"Foo ({self.name}): __init__")

    def __del__(self):
        print(f"Foo ({self.name}): __del__")

f = Foo("Aston")
g = Foo("Lukas")

del f

x = input("Wpisz coś: ")

something()

y = []

print(y[0])
print("Jestem tutaj")

File: programowanie obiektowe/2025.03.28/c31.cpp
#include <iostream>

class Foo {
public:
    Foo(int x) : x(x){
        std::cout << "Foo" << std::endl;
    }

    ~Foo(){
        std::cout << "~Foo" << std::endl;
    }
    int x;
};

int main() {
    Foo f = Foo(123);

    int x=2, y=0;
    int z=x/y;

    std::cout << x/y << std::endl;
    return 0;
}